src/Geom_Graph.f90

```diff
@@ -645,16 +820,16 @@ PURE SUBROUTINE getMidPointOnEdge_graphType(thisGraph,v1,v2,m)
   REAL(SRK) :: a(2),b(2),c(2),r,alp1,alp2,theta,scal
 
   m=-HUGE(m)
-  n=nVert_graphType(thisGraph)+1
+  n=this%nVert()+1
   IF(v1 > 0 .AND. v2 > 0 .AND. v1 < n .AND. v2 < n) THEN
-    IF(thisGraph%edgeMatrix(v1,v2) == 1) THEN
-      m=thisGraph%vertices(:,v1)+thisGraph%vertices(:,v2)
+    IF(this%edgeMatrix(v1,v2) == 1) THEN
```

Comment:
Looks like you can use the enumerations here.

Reply:
good catch.  I need to do another pass for enumerations... unfortunately searching for 1, -1, and 0 is kind of annoying

Reply:
fixed as many of these as I could find